tests/src/python/test_qgsaggregatecalculator.py

```diff
@@ -55,6 +60,31 @@ def testParameters(self):
         self.assertEqual(a.filter(), 'string filter')
         self.assertEqual(a.delimiter(), 'delim')
 
+    def testGeometry(self):
+        """ Test calculation of aggregates on geometry expressions """
+
+        layer = QgsVectorLayer("Point?",
+                               "layer", "memory")
+        pr = layer.dataProvider()
+
+        # must be same length:
+        geometry_values = [QgsGeometry.fromWkt("Point ( 0 0 )"), QgsGeometry.fromWkt("Point ( 1 1 )"), QgsGeometry.fromWkt("Point ( 2 2 )")]
+
+        features = []
+        for i in range(len(geometry_values)):
+            f = QgsFeature()
+            f.setGeometry(geometry_values[i])
+            features.append(f)
+        assert pr.addFeatures(features)
+
+        agg = QgsAggregateCalculator(layer)
+
+        val, ok = agg.calculate(QgsAggregateCalculator.GeometryCollect, '$geometry')
+        self.assertTrue(ok)
+        expwkt = "MultiPoint ((0 0), (1 1), (2 2))"
+        wkt = val.exportToWkt()
+        assert compareWkt(expwkt, wkt), "Expected:\n%s\nGot:\n%s\n" % (expwkt, wkt)
+
```

Comment:
Can you add 

```
[QgsAggregateCalculator.GeometryCollect, 'collect']
```

to the tests in testStringToAggregate

and something like

```
        # geometry
        val, ok = agg.calculate(QgsAggregateCalculator.GeometryCollect, "make_point( coalesce(fldint,0), 2 )")
        self.assertTrue(ok)
        self.assertTrue(val.exportToWkt(), 'MultiPoint((4 2, 2 2, 3 2, 2 2,5 2, 0 2,8 2))')
```

to testExpression